From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: ctcm: avoid crash in ctcm_remove_device
References: bnc#487755,LTC#52679

Symptom:     crash in debugging call of ctcm_remove_device
Problem:     Channels are already removed when setting a
             ctcm-device offline. Thus ctcm_remove_device must
             not refer to channel information.
Solution:    delete channel information from the trace call in
             ctcm_remove_device.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/net/ctcm_main.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

Index: linux-sles11/drivers/s390/net/ctcm_main.c
===================================================================
--- linux-sles11.orig/drivers/s390/net/ctcm_main.c
+++ linux-sles11/drivers/s390/net/ctcm_main.c
@@ -1665,10 +1665,8 @@ static void ctcm_remove_device(struct cc
 	BUG_ON(priv == NULL);
 
 	CTCM_DBF_TEXT_(SETUP, CTC_DBF_INFO,
-			"removing device %s, r/w = %s/%s, proto : %d",
-			priv->channel[READ]->netdev->name,
-			priv->channel[READ]->id, priv->channel[WRITE]->id,
-			priv->protocol);
+			"removing device %p, proto : %d",
+			cgdev, priv->protocol);
 
 	if (cgdev->state == CCWGROUP_ONLINE)
 		ctcm_shutdown_device(cgdev);
